vectorial_model.py

```diff
@@ -0,0 +1,141 @@
+    filtered_query = remove_stop_words(filtered_query, stop_words)
+    normalized_query = lemmatize(filtered_query)
+    return normalized_query
+
+
+def get_term_frequency(term, doc_id, index_frequency):
+    """Get the frequency of a term in a document of the collection."""
+    return index_frequency[term][doc_id]
+
+
+def get_logarithmic_term_frequency(term, doc_id, index_frequency):
+    """
+    Get logarithmic frequency of term in a document of the collection.
+
+    Equal to 0 if the term is not in the document, otherwise 1 + log(term_frequency).
+    """
+    tf = get_term_frequency(term, doc_id, index_frequency)
+    if tf > 0:
+        return 1 + math.log(tf)
+    else:
+        return 0
+
+
+def get_normalized_term_frequency(term, doc_id, index_frequency, stats_collection):
+    """Get normalized term frequency"""
+    term_frequency = get_term_frequency(term, doc_id, index_frequency)
+    normalized_term_frequency = 0.5 + 0.5 * (term_frequency / stats_collection[doc_id]["max_frequency"])
+    return normalized_term_frequency
+
+
+def get_normalized_logarithmic_term_frequency(term, doc_id, index_frequency, stats_collection):
+    """Get normalized logarithmic term frequency"""
+    term_frequency = get_term_frequency(term, doc_id, index_frequency)
+    normalized_logarithmic_term_frequency = (1 + math.log(term_frequency)) / (1 + math.log(
+        stats_collection[doc_id]["average_frequency"]))
+    return normalized_logarithmic_term_frequency
+
+
+def get_idf(term, index_frequency, nb_doc):
+    """Gets idf of term in collection"""
+    return math.log(nb_doc / len(index_frequency[term].keys()))
+
+
+def get_stats_document(document):
+    """
+    Get statistics on the document.
+
+    Calculates the frequency of most common term, the number of unique terms, and the average frequency of terms.
+    :param document: Document to get statistics on
+    :return: (dict) stats
+    """
+    counter = Counter()
+    for term in document:
+        counter.update([term])
+    stats = dict()
+    stats["max_frequency"] = counter.most_common(1)[0][1]
+    stats["number_unique_terms"] = len(counter.items())
+    tf_moy = sum(counter.values())
+    stats["average_frequency"] = tf_moy / len(counter.items())
+    return stats
+
+
+def get_stats_collection(collection):
+    """Get statistics on all the documents in the collection """
+    stats = dict()
+    stats["nb_docs"] = len(collection.keys())
+    for doc in collection:
+        stats[doc] = get_stats_document(collection[doc])
+    return stats
+
+
+def process_vectorial_query(query, frequency_index, stop_words, stats_collection, weighting_scheme_document,
+                            weighting_scheme_query):
+    # Set weighting schemes if not specified
+    if weighting_scheme_document is None:
+        weighting_scheme_document = "tf_idf_normalize"
+    if weighting_scheme_query is None:
+        weighting_scheme_query = "frequency"
```

Comment:
You should put this as default values in the params, to make them optional

Reply:
If I understand what you mean correctly that doesn’t work as the value from the parser would still be None, I changed in the argument parser instead.

Reply:
@owain-Biddulph actually you also should put it in as default the params of the function because the function could be called from somewhere else with None values

Reply:
I’ll add it in, but it still only works if the arguments aren’t given, not if None values are passed.